sanic/worker.py

```diff
@@ -109,7 +114,13 @@ def run(self):
             while self.alive:
                 self.notify()
 
-                if pid == os.getpid() and self.ppid != os.getppid():
+                req_count = sum(
+                    self.servers[srv]["requests_count"] for srv in self.servers
+                )
+                if self.cfg.max_requests and req_count > self.cfg.max_requests:
```

Comment:
It's better to use `self.max_requests` to support 'jitter'.

https://github.com/benoitc/gunicorn/blob/master/gunicorn/workers/base.py#L54